api/src/services/user.js

```diff
@@ -162,6 +162,37 @@ async function softDeleteUser(username) {
   return updatedUser ? transformUser(updatedUser) : updatedUser;
 }
 
+async function hardDeleteUser(username) {
+  if (!username) {
+    throw new Error("Username is required for deletion.");
+  }
+
+  try {
+    console.log(`Starting deletion process for username: ${username}`);
+
+    // Find the user by username
+    const user = await prisma.user.findUnique({ where: { username } });
+    if (!user) {
+      throw new Error(`User with username "${username}" not found.`);
+    }
+    
+    // Store user details for logging purposes
+    const userDetails = { id: user.id, username: user.username };
+    // Delete the user (cascading delete and nullifications are handled by the schema)
+    await prisma.user.delete({ where: { id: user.id } });
+
```

Comment:
Try to use a single Prisma transaction. This can be done thus:

```
await prisma.delete(
  { where: username: 'someUsername' }
)
```

The `username` field has a `@unique` constraint, so using prisma's `delete` method will find and delete the record corresponding to that username only.